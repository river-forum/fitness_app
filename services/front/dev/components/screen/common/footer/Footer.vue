<template>
  <div
    class="flex items-center justify-between px-8 py-5 max-w-md relative"
    style="background-color: #3b3b3b"
  >
    <div
      class="absolute text-white rounded-full bg-green-500 cursor-pointer"
      style="top: 0; left: 50%; transform: translate(-50%, -50%)"
    >
      <img
        class="m-3 hover:opacity-75"
        src="@/assets/icons/add.svg"
        width="20"
        height="20"
        alt="add"
      />
    </div>
    <img
      class="cursor-pointer hover:opacity-75"
      :class="active('grid')"
      src="@/assets/icons/grid.svg"
      width="25"
      height="25"
      alt="grid"
      @click="clickListener('grid')"
    />
    <img
      class="cursor-pointer hover:opacity-75"
      :class="active('dumbbell')"
      src="@/assets/icons/dumbbell.svg"
      width="25"
      height="25"
      alt="dumbbell"
      @click="clickListener('dumbbell')"
    />
    <img
      class="cursor-pointer hover:opacity-75"
      :class="active('graph')"
      src="@/assets/icons/graph.svg"
      width="25"
      height="25"
      alt="graph"
      @click="clickListener('graph')"
    />
    <img
      class="cursor-pointer hover:opacity-75"
      :class="active('calendar')"
      src="@/assets/icons/calendar.svg"
      width="25"
      height="25"
      alt="calendar"
      @click="clickListener('calendar')"
    />
  </div>
</template>
<script>
import { defineComponent, ref, computed } from '@nuxtjs/composition-api'

/**
 * #Footerコンポーネント(SP幅専用)
 */
export default defineComponent({
  setup() {
    const current = ref('grid')
    const active = computed(() => (val) =>
      current.value === val ? 'opacity-100' : 'opacity-25'
    )

    function clickListener(val) {
      current.value = val
    }

    return {
      clickListener,
      active,
    }
  },
})
</script>
