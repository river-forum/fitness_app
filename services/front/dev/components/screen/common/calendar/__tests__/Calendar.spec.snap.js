import { mount } from '@vue/test-utils'
import Footer from '../Calendar.vue'

it('Calendar Component renders correctly', () => {
  const app = mount(Footer)

  expect(app.html()).toMatchSnapshot()
})
