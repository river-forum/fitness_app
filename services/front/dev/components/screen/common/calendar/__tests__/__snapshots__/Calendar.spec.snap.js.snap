// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Component renders correctly 1`] = `
"<div class=\\"max-w-md calendar\\">
  <div class=\\"relative p-4 pb-0 bg-green-500 calendar__wrap\\">
    <div class=\\"absolute top-0 right-0 ymd\\">
      <ul class=\\"flex ymd__contents\\">
        <li class=\\"opacity-75 ymd__li\\"><button class=\\"px-4 py-1 font-bold text-white bg-black\\">
            D
          </button></li>
        <li class=\\"opacity-75 ymd__li\\"><button class=\\"px-4 py-1 font-bold text-white bg-black\\">
            W
          </button></li>
        <li class=\\"opacity-75 ymd__li\\"><button class=\\"px-4 py-1 font-bold text-white bg-black\\">
            M
          </button></li>
      </ul>
    </div>
    <div class=\\"text-white date\\">
      <div>
        <div class=\\"date__current-month font-bold\\">

        </div>
        <ul class=\\"date__day flex overflow-scroll\\">
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">22</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">23</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">24</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">25</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">26</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">27</span> <br><span class=\\"day__of-week\\">火</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">28</span> <br><span class=\\"day__of-week\\">水</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">29</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">30</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">31</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">1</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">2</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">3</span> <br><span class=\\"day__of-week\\">火</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">4</span> <br><span class=\\"day__of-week\\">水</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">5</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">6</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">7</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">8</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">9</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">10</span> <br><span class=\\"day__of-week\\">火</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">11</span> <br><span class=\\"day__of-week\\">水</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">12</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">13</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">14</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">15</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">16</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">17</span> <br><span class=\\"day__of-week\\">火</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">18</span> <br><span class=\\"day__of-week\\">水</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">19</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">20</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">21</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">22</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">23</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">24</span> <br><span class=\\"day__of-week\\">火</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">25</span> <br><span class=\\"day__of-week\\">水</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">26</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">27</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">28</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">29</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">30</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">1</span> <br><span class=\\"day__of-week\\">火</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">2</span> <br><span class=\\"day__of-week\\">水</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">3</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">4</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">5</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">6</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">7</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">8</span> <br><span class=\\"day__of-week\\">火</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">9</span> <br><span class=\\"day__of-week\\">水</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">10</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">11</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">12</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">13</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">14</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">15</span> <br><span class=\\"day__of-week\\">火</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">16</span> <br><span class=\\"day__of-week\\">水</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">17</span> <br><span class=\\"day__of-week\\">木</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">18</span> <br><span class=\\"day__of-week\\">金</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">19</span> <br><span class=\\"day__of-week\\">土</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">20</span> <br><span class=\\"day__of-week\\">日</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">21</span> <br><span class=\\"day__of-week\\">月</span></button></li>
          <li class=\\"day\\"><button class=\\"day__btn\\"><span class=\\"day__num\\">22</span> <br><span class=\\"day__of-week\\">火</span></button></li>
        </ul>
      </div>
    </div>
  </div>
</div>"
`;
